### PR TITLE
Feature/ssl refactor

### PR DESCRIPTION
**Pull request by caiola**

### CODE REVIEW

File: backend/dockerfile-apiserver-dev [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ARG PYTHON_VERSION
```
Comment (sergiodenboer) [inline backend/dockerfile-apiserver-dev:1]: No need for this argument.

File: backend/dockerfile-apiserver-local [hunk 2]
```diff
@@ -0,0 +1,20 @@
+ARG PYTHON_VERSION
```
Comment (sergiodenboer) [inline backend/dockerfile-apiserver-local:1]: no need for this argument.

File: backend/dockerfile-apiserver-local [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Make a working directory in the image and set it as working dir.
+RUN mkdir /app
+WORKDIR /app
+
+# make sure that pip & setuptools are installed and to date
+RUN pip install --upgrade pip setuptools wheel && pip install poetry
+
+# ERROR "you need a C compiler to build uWSGI"
+# ENV local does not need it
+# RUN apt-get -y install gcc
```
Comment (sergiodenboer) [inline backend/dockerfile-apiserver-local:13]: Isn't gcc already in the pytohn images?

File: backend/server.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from flasgger import Swagger
+from flask import Flask
+from flask.blueprints import Blueprint
+
+import config
+import routes
+from models import db
+
+# config your API specs
```
Comment (sergiodenboer) [inline backend/server.py:9]: No need for this comment block.

File: docker-compose-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+    environment:
+      TZ: ${TIMEZONE}
+    restart: unless-stopped
+    networks:
+      - netpublic
+
+  apiserver:
+    hostname: apiserver
+    #environment:
```
Comment (sergiodenboer) [inline docker-compose-deploy.yml:30]: Please delete comented fields.

File: docker-compose-dev.yml [hunk 6]
```diff
@@ -10,15 +10,14 @@ services:
     ports:
       - 80:80
       - 443:443
-    container_name: traefik
     hostname: traefik
+    container_name: traefik
```
Comment (sergiodenboer) [inline docker-compose-dev.yml:14]: Contianer name and hostname equal by default?

File: docker-compose-dev.yml [hunk 7]
```diff
@@ -10,15 +10,14 @@ services:
...
     environment:
       - DOMAIN_VHOST_ROOT=${DOMAIN_VHOST_ROOT}
```
Comment (sergiodenboer) [inline docker-compose-dev.yml:16]: Quite the list. Shound't this be moved to a file instead?

File: docker-compose-dev.yml [hunk 8]
```diff
@@ -118,52 +98,69 @@ services:
...
+      - "traefik.http.routers.appserver.tls.certresolver=le"
+      - "traefik.http.routers.appserver.tls.domains[0].main=${DOMAIN_VHOST_APP}"
     networks:
       - netpublic
 
-  testdb:
-    image: oraclelinux:8-slim
-    networks:
-      - netpublic
+#  webserver:
```
Comment (sergiodenboer) [inline docker-compose-dev.yml:124]: Please remove comented code.

File: docker-compose-prod.yml [hunk 9]
```diff
@@ -32,6 +31,11 @@ services:
       - --certificatesresolvers.le.acme.email=${TRAEFIK_SSLEMAIL?Variable not set}
       - --certificatesresolvers.le.acme.storage=./acme.json
       - --certificatesresolvers.le.acme.tlschallenge=true
+      # - "--certificatesresolvers.le.acme.caserver=https://acme-staging-v02.api.le.org/directory"
```
Comment (sergiodenboer) [inline docker-compose-prod.yml:34]: Remove commented code.

File: backend/dockerfile-apiserver-local (hunk 3 above)
Comment (caiola) [reply backend/dockerfile-apiserver-local:13]: I tried to use the slim version of the python image but gave issues with the SQL migrations.
It is a comment to improve later to the slim version if the migrations are properly fixed.

File: backend/dockerfile-apiserver-dev (hunk 1 above)
Comment (caiola) [reply backend/dockerfile-apiserver-dev:1]: It is to be able to customize the version of the image through variables

**Pull request merged**